librad/src/git/types.rs

```diff
@@ -153,6 +153,118 @@ impl<'a> Into<ext::blob::Branch<'a>> for &'a Reference {
     }
 }
 
+pub struct SymbolicReference<'a> {
+    repo: &'a git2::Repository,
+}
+
+pub struct SymbolicReferenceSource<'a> {
+    repo: &'a git2::Repository,
+    source: String,
+}
+
+enum SymbolicReferenceTargetArgument<'a> {
```

Comment:
I think this is optimising for a case we don't even have yet. So my feeling is that it's not necessary for now and we solve for the simplest case of have two `librad::Reference`s.